pandas/tests/extension/base/setitem.py

```diff
@@ -93,6 +93,90 @@ def test_setitem_iloc_scalar_multiple_homogoneous(self, data):
         df.iloc[10, 1] = data[1]
         assert df.loc[10, "B"] == data[1]
 
+    @pytest.mark.parametrize(
+        "mask",
+        [
+            np.array([True, True, True, False, False]),
+            pd.array([True, True, True, False, False], dtype="boolean"),
```

Comment:
add a list, maybe even add Index-object dtype of bools

Reply:
>  add Index-object dtype of bools

That is explicitly not supported in general (and that the validation raises an error for that is already tested elsewhere)